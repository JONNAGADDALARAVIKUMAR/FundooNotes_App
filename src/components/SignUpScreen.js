import React, { Component } from 'react';
import {Image, ScrollView, TextInput, View, Text, TouchableOpacity} from 'react-native';
import SignUpStyles from '../styles/SignUpStyles';

export default class LogInScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            AppName: props.AppName,
            ScreenType: props.ScreenType,
            email: '',
            password: '',
            firstName: '',
            lastName: '',
            confirmPassword: '',
            emailValidation: true,
            firstNameValidation: true,
            lastNameValidation: true,
            passwordValidation: true,
            passwordSecurity: true,
            confirmPasswordSecurity: true,
        }
    }
         
    firstNameHandler = async (enteredFirstName) => {  
        await this.setState ({
            firstName: enteredFirstName
        })
        console.log(this.state.firstName);
        const nameRejex = new RegExp("(^[A-Za-z]+$)");
        (nameRejex.test(this.state.firstName)) ? this.setState({firstNameValidation: true}) : this.setState({firstNameValidation: false}) 
    }

    lastNameHandler = async (enteredLastName) => {
        await this.setState({
            lastName: enteredLastName
        })
        const nameRejex = new RegExp("(^[A-Za-z]+$)");
        (nameRejex.test(this.state.lastName)) ? this.setState({lastNameValidation: true}) : this.setState({lastNameValidation: false})
    }

    emailHandler = async (enteredEmail) => {
        await this.setState({
            email: enteredEmail
        })
        this.checkEmail();
    }
    checkEmail = async () => {
        const emailRejex = new RegExp("^[0-9a-zA-Z]+([._+-][0-9A-Za-z]+)*@[0-9A-Za-z]+[.][a-zA-Z]{2,4}([.][a-zA-Z]{2,4})?$");       
        if(emailRejex.test(this.state.email)) {
            this.setState({
                emailValidation: true
            })
        }
        else {
            this.setState({
                emailValidation: false
            })
        }
    }

    passwordHandler = async (enteredPassword) => {
        await this.setState({
            password: enteredPassword
        })
        this.checkPassword();
    }
    checkPassword = async () => {
        const strongRegex = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");
        if(strongRegex.test(this.state.password)) {
            this.setState({
                passwordValidation: true
            })
        }
        else {
            this.setState({
                passwordValidation: false
            })
        }
    }

    confirmPasswordHandler = async (enteredPassword) => {
        await this.setState({
            confirmPassword: enteredPassword
        })
    }
    passwordSecurityHandler = async () => {
        //const {onPress} = this.props
        var passwordSecurity = this.state.passwordSecurity
        if(passwordSecurity == true) {
            this.setState({
                passwordSecurity: false
            })
        }
        else {
            this.setState({
                passwordSecurity: true
            })
        }
        //onPress();
    }
    confirmPasswordSecurityHandler = async () => {
        //const {onPress} = this.props
        var confirmPasswordSecurity = this.state.confirmPasswordSecurity
        if(confirmPasswordSecurity == true) {
            this.setState({
                confirmPasswordSecurity: false
            })
        }
        else {
            this.setState({
                confirmPasswordSecurity: true
            })
        }
        //onPress();
    }

    render() {
        return (
            <View style = {SignUpStyles.background_Styles}>
                <Text style = {SignUpStyles.heading_Style}>{this.state.ScreenType}</Text>
                <ScrollView>
                    <Image style = {SignUpStyles.Logo_Style}
                        source = {require('../assets/FundoIcon.png')}
                    />
                    <Text style = {SignUpStyles.Fundo_Style}>{this.state.AppName}</Text>

                    <View>
                        <TextInput
                            style = {SignUpStyles.Input_TextBox_Style}
                            placeholder = "First Name"
                            placeholderTextColor = "#b0686d"
                            onChangeText = {(firstName) => this.firstNameHandler(firstName.toUpperCase())}
                        />
                        <Text style = {SignUpStyles.pop_up_Message}>
                            {(this.state.firstNameValidation || this.state.firstName == '') ? null : 'Invalid First Name..'}
                        </Text>

                        <TextInput
                            style = {SignUpStyles.Input_TextBox_Style}
                            placeholder = "Last Name"
                            placeholderTextColor = "#b0686d"
                            onChangeText = {(lastName) => this.lastNameHandler(lastName)}
                        />
                        <Text style = {SignUpStyles.pop_up_Message}>
                            {(this.state.lastNameValidation || this.state.lastName == '') ? null : 'Invalid Last Name..'}
                        </Text>

                        <TextInput
                            style = {SignUpStyles.Input_TextBox_Style}
                            placeholder = "Email"
                            placeholderTextColor = "#b0686d"
                            onChangeText = {(email) => this.emailHandler(email)}
                        />
                        <Text style = {SignUpStyles.pop_up_Message}>
                            {(this.state.emailValidation || this.state.email == '') ? null : 'Invalid Email..'}
                        </Text>

                        <View style = {[SignUpStyles.Input_TextBox_Style, SignUpStyles.set_icon]}>
                            <TextInput style = {{width: '88%'}}
                                placeholder =   "Password"
                                maxLength = {25}
                                placeholderTextColor = "#b0686d"
                                secureTextEntry = {this.state.passwordSecurity}
                                onChangeText = {(password) => this.passwordHandler(password)}
                            />
                            {(this.state.passwordSecurity) ?
                                <TouchableOpacity 
                                    style= {{alignSelf: 'center'}}
                                    onPress = {this.passwordSecurityHandler}>
                                        <Image style = {SignUpStyles.icon} source = {require('../assets/showPasswordIcon.png')}/>
                                </TouchableOpacity> 
                                : 
                                <TouchableOpacity 
                                    style = {{alignSelf: 'center'}}
                                    onPress = {this.passwordSecurityHandler}>
                                        <Image style = {SignUpStyles.icon} source = {require('../assets/hidePasswordIcon.png')}/>
                                </TouchableOpacity>  
                            }
                        </View>
                        <Text style = {SignUpStyles.pop_up_Message}>
                            {(this.state.passwordValidation || this.state.password == '') ? null : 'Weak password..'}
                        </Text>

                        <View style = {[SignUpStyles.Input_TextBox_Style, SignUpStyles.set_icon]}>
                            <TextInput style = {{width: '88%'}}
                                placeholder =   "Confirm Password"
                                maxLength = {25}
                                placeholderTextColor = "#b0686d"
                                secureTextEntry = {this.state.confirmPasswordSecurity}
                                onChangeText = {(confirmPassword) => this.confirmPasswordHandler(confirmPassword)}
                            />
                            {(this.state.confirmPasswordSecurity) ?
                                <TouchableOpacity 
                                    style= {{alignSelf: 'center'}}
                                    onPress = {this.confirmPasswordSecurityHandler}>
                                        <Image style = {SignUpStyles.icon} source = {require('../assets/showPasswordIcon.png')}/>
                                </TouchableOpacity> 
                                : 
                                <TouchableOpacity 
                                    style = {{alignSelf: 'center'}}
                                    onPress = {this.confirmPasswordSecurityHandler}>
                                        <Image style = {SignUpStyles.icon} source = {require('../assets/hidePasswordIcon.png')}/>
                                </TouchableOpacity>  
                            }
                        </View>
                        <Text style = {SignUpStyles.pop_up_Message}>
                            {(this.state.password.includes(this.state.confirmPassword)) ? null : 'Password MissMatch'}
                        </Text>
                    </View>
                    <View style = {{flexDirection: 'row'}}>
                        <Text style = {SignUpStyles.Have_Account_Style}>Already Have Account</Text>
                        <TouchableOpacity>
                            <Text style = {SignUpStyles.ClickHere_Style}>Click Here</Text>
                        </TouchableOpacity>
                    </View>
                    <View>
                        <TouchableOpacity style = {[SignUpStyles.SignUp_Button_Styles, SignUpStyles.Button_Styles]}>
                            <Text style = {{color: '#dbced2'}}>SIGN UP</Text>
                        </TouchableOpacity>
                    </View>
                </ScrollView>
            </View>
        )
    }
}