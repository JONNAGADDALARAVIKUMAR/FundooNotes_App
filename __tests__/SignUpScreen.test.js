import React from 'react';
import {shallow} from 'enzyme';

import SignUpScreen from '../src/components/SignUpScreen'

describe('test SignUp', () => {
    it('test when render should match to snapshot', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component).toMatchSnapshot();
    })
    it('test the firstName handler method should update firstName state', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().state.firstName).toBe('')
        component.instance().firstNameHandler('Ravi')
        expect(component.instance().state.firstName).toBe('RAVI')
    })
    it('test the lastName handler method should update lastName state', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().state.lastName).toBe('')
        component.instance().lastNameHandler('Kumar')
        expect(component.instance().state.lastName).toBe('KUMAR')
    })
    it('test the email handler method should update email state', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().state.email).toBe('')
        component.instance().emailHandler('kumar@gmail.com')
        expect(component.instance().state.email).toBe('kumar@gmail.com')
    })
    it('test the password handler method should update password state', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().state.password).toBe('')
        component.instance().passwordHandler('Ravi@765')
        expect(component.instance().state.password).toBe('Ravi@765')
    })
    it('test the Confirm password handler method should update password state', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().state.confirmPassword).toBe('')
        component.instance().confirmPasswordHandler('Ravi@765')
        expect(component.instance().state.confirmPassword).toBe('Ravi@765')
    })
    it('test the email handler method should update emailValidation state true', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().setState({email: 'ravi@gmail.com'}))
        component.instance().checkEmail()
        expect(component.instance().state.emailValidation).toBe(true)//Happy Test case 
    })
    it('test the email handler method should update emailValidation state False', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().setState({email: 'ravi@gmail'}))
        component.instance().checkEmail()
        expect(component.instance().state.emailValidation).toBe(false)//Sad Test case 
    })
    it('test the Password handler method should update passwordValidation state true', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().setState({password: 'Ravi@433'}))
        component.instance().checkPassword()
        expect(component.instance().state.passwordValidation).toBe(true)//Happy Test case 
    })
    it('test the Password handler method should update passwordValidation state False', async () => {
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'}/>)
        expect(component.instance().setState({password: 'ravi@433'}))
        component.instance().checkPassword()
        expect(component.instance().state.passwordValidation).toBe(false)//Sad Test case 
    })
    
    it('test onPress event of eye icon of password textinput called it will change the passwordSecurity State', async () => {
        const onPressEvent = jest.fn();
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'} onPress = {onPressEvent}/>)
        const instance = component.instance();
        expect(instance.state.passwordSecurity).toBe(true);
        instance.passwordSecurityHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(1);
        expect(instance.state.passwordSecurity).toBe(false);
        instance.passwordSecurityHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(2);
        expect(instance.state.passwordSecurity).toBe(true);
    })
    it('test onPress event of eye icon of password textinput called it will change the confirmPasswordSecurity State', async () => {
        const onPressEvent = jest.fn();
        const component = shallow(<SignUpScreen AppName = "Fundo Notes" ScreenType = {'Create Account'} onPress = {onPressEvent}/>)
        const instance = component.instance();
        expect(instance.state.confirmPasswordSecurity).toBe(true);
        instance.confirmPasswordSecurityHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(1);
        expect(instance.state.confirmPasswordSecurity).toBe(false);
        instance.confirmPasswordSecurityHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(2);
        expect(instance.state.confirmPasswordSecurity).toBe(true);
    })
})