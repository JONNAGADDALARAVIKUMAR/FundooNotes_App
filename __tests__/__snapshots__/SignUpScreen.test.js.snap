// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SignUp test when render should match to snapshot 1`] = `ShallowWrapper {}`;
